include/seastar/util/print_safe.hh

```diff
@@ -84,14 +84,14 @@ char* convert_hex_safe(char *buf, size_t bufsz, Integral n) noexcept {
 template<typename Integral>
 SEASTAR_CONCEPT( requires std::integral<Integral> )
 void convert_zero_padded_hex_safe(char *buf, size_t bufsz, Integral n) noexcept {
-    convert_hex_safe<'0'>(buf, bufsz, n);
+    convert_hex_safe<Integral, '0'>(buf, bufsz, n);
```

Comment:
why? i think the compiler should be able to figure out the first template paremeter for `convert_hex_safe<>`, right?

Reply:
No. How is the compiler supposed to know that `0` is the *second* template argument?

Only trailing arguments can be deduced.